packages/zoltan2/test/color/ReadMatrixFromFile.hpp

```diff
@@ -0,0 +1,319 @@
+#ifndef __READMATRIXFROMFILE_HPP
+#define __READMATRIXFROMFILE_HPP
+#include<vector>
+template <typename global_ordinal_type, typename local_ordinal_type, typename scalar_type, typename map_type>
+void
+distribute (Teuchos::ArrayRCP<size_t>& myNumEntriesPerRow,
+	    Teuchos::ArrayRCP<size_t>& myRowPtr,
+	    Teuchos::ArrayRCP<global_ordinal_type>& myColInd,
+	    Teuchos::ArrayRCP<scalar_type>& myValues,
+	    const Teuchos::RCP<const map_type>& pRowMap,
+	    global_ordinal_type *rowPtr,
+	    global_ordinal_type *colInd,
+	    const bool debug=false)
+{
+  
+  Teuchos::RCP<const Teuchos::Comm<int>> pComm = pRowMap->getComm ();
+  const int numProcs = pComm->getSize ();
+  const int myRank = pComm->getRank ();
+  const int rootRank = 0;
+
+  Teuchos::ArrayView<const global_ordinal_type> myRows = pRowMap->getNodeElementList();
+  const size_t myNumRows = myRows.size();
+  
+  myNumEntriesPerRow = Teuchos::ArrayRCP<size_t> (myNumRows);
+
+  if (myRank != rootRank) {
+
+    Teuchos::send (*pComm, myNumRows, rootRank);
+    
+    if (myNumRows != 0) {
+
+      Teuchos::send (*pComm, static_cast<int> (myNumRows), myRows.getRawPtr(), rootRank);      
+      Teuchos::receive (*pComm, rootRank, static_cast<int> (myNumRows), myNumEntriesPerRow.getRawPtr());
+
+      const local_ordinal_type myNumEntries =
+	std::accumulate (myNumEntriesPerRow.begin(),
```

Comment:
This is fine for now, but be aware that `std::accumulate`'s internal accumulation type is the type of the third argument, which is `int` here (the type of `0`).  That would be bad if you were summing up 64-bit integers, for example.

Reply:
I will make sure to check that that doesn't pose any potential issues.

Reply:
@ibogle Canonical way would be to cast the initial value to `local_ordinal_type`.
```cpp
const local_ordinal_type result =
  std::accumulate(x.begin(), x.end(), local_ordinal_type(0));
```